contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol

```diff
@@ -141,6 +141,58 @@ contract KeeperRegistrar2_0 is TypeAndVersionInterface, ConfirmedOwner, ERC677Re
     uint8 source,
     address sender
   ) external onlyLINK {
+    _register(name, encryptedEmail, upkeepContract, gasLimit, adminAddress, checkData, amount, source, sender);
+  }
+
+  /**
+   * @notice Allows external users to register upkeeps
+   * @notice registerUpkeep can only be called through transferAndCall on LINK contract
+   * @param name string of the upkeep to be registered
+   * @param encryptedEmail email address of upkeep contact
+   * @param upkeepContract address to perform upkeep on
+   * @param gasLimit amount of gas to provide the target contract when performing upkeep
+   * @param adminAddress address to cancel upkeep and withdraw remaining funds
+   * @param checkData data passed to the contract when checking for upkeep
+   * @param amount quantity of LINK upkeep is funded with (specified in Juels)
+   * @param source application sending this request
+   */
+  function registerUpkeep(
+    string memory name,
+    bytes calldata encryptedEmail,
+    address upkeepContract,
+    uint32 gasLimit,
+    address adminAddress,
+    bytes calldata checkData,
+    uint96 amount,
+    uint8 source
+  ) external returns (uint256) {
+    if (amount < s_config.minLINKJuels) {
+      revert InsufficientPayment();
+    }
+
+    KeeperRegistryBaseInterface keeperRegistry = s_config.keeperRegistry;
+
+    bool success = LINK.transferFrom(msg.sender, owner(), amount);
```

Comment:
LINK.transferFrom(msg.sender,address(this), amount);

Reply:
interesting, I see here https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/KeeperRegistry1_2.sol#L297-L303 that we don't care about return value. Can you dig a bit into if its really needed

Reply:
![image](https://user-images.githubusercontent.com/8682222/187767124-72785029-1485-4e8e-997e-6766d0fe45ea.png)
 
looks like we shouldn't have to worry about it.